TAScheduler/UnitTests/tests.py

```diff
@@ -46,38 +52,476 @@ class TestAdminCreateSection(TestCase):  # Alec
+                         self.hold_course, "Should have linked course and instructor together")
+
+    def test_max_capacity_instr(self):
+        User.delete(self.hold_user)
+        temp = User.objects.create(self.hold_user)
+        instr = Instructor.objects.create(
+            user=temp,
+            max_assignments=0
+        )
+        temp_in = InstructorObj(instr)
+        with self.assertRaises(RuntimeError, msg="Tried to link course to instructor with max assignments"):
+            self.admin.courseInstrAsmgt(temp_in, self.tempCourse)
 
 
 class TestAdminCourseTAAsgmt(TestCase):  # Kevin
-    pass
+    tempCourse = None
+    tempTA = None
+    admin = None
+    hold_course = None
+    hold_user = None
+
+    def setUp(self):
+        self.hold_course = Course.objects.create(
+            course_id=101,
+            semester='Fall 2023',
+            name='Introduction to Testing',
+            description='A course about writing tests in Django.',
+            num_of_sections=3,
+            modality='Online',
+            credits=4
+        )
+        self.tempCourse = CourseObj(self.hold_course)
+        self.hold_user = User.objects.create(
+            email_address='kev@example.com',
+            password='kevpassword',
+            first_name='Kevin',
+            last_name='User',
+            home_address='123 Kevin St',
+            phone_number='1234667890'
+        )
+        temp_ta = TA.objects.create(user=self.hold_user, grader_status=True)
+        self.tempTA = TAObj(temp_ta)
+        hold_user = User(
+            email_address='admin@example.com',
+            password='adminpassword',
+            first_name='Admin',
+            last_name='User',
+            home_address='123 Admin St',
+            phone_number='1234567890'
+        )
+        hold_user.save()
+        hold_admin = Administrator(user=hold_user)
+        hold_admin.save()
+        self.admin = AdminObj(hold_admin)
+
+    def test_bad_instr(self):
+        with self.assertRaises(TypeError, msg='TA that was passed is not a valid TA'):
+            self.admin.courseInstrAsmgt(11, self.tempCourse)
+
+    def test_bad_course(self):
+        with self.assertRaises(TypeError, msg='Course that was passed is not a valid course'):
+            self.admin.courseInstrAsmgt(self.tempTA, 11)
+
+    def test_null_instr(self):
+        User.delete(self.hold_user)
+        with self.assertRaises(RuntimeError, msg="Tried to link course to non existant user"):
+            self.admin.courseTAAsmgt(self.tempTA, self.tempCourse)
+
+    def test_null_course(self):
+        Course.delete(self.hold_course)
+        with self.assertRaises(RuntimeError, msg="Tried to link user to non existant course"):
+            self.admin.courseTAAsmgt(self.tempTA, self.tempCourse)
+
+    def test_success_connect(self):
+        self.admin.courseTAAsmgt(self.tempTA, self.tempCourse)
+        self.assertEqual(TAToCourse.objects.get(course=self.hold_course).course,
```

Comment:
.get(course=..., ta = ...) pretty sure it needs the "ta=" for reasons given previously